lib/streamlit/elements/widgets/chat.py

```diff
@@ -194,7 +197,9 @@ def chat_message(
         ):
             # For selected labels, we are mapping the label to an avatar
             avatar = name.lower()
-        avatar_type, converted_avatar = _process_avatar_input(avatar)
+        avatar_type, converted_avatar = _process_avatar_input(
+            avatar, self.dg._get_delta_path_str()
```

Comment:
@LukasMasuch one question I have, since I am not super familiar with the `chat_message` design.
Is `self.dg._get_delta_path_str()`return unique "coordinates" for different chat_messages?

I looked into the `media_file_manager` implementation and looks like this should not cause "multiple uploads of the same file"  / "unwanted premature file removals for the file" issues. 

Just to confirm that everything works correctly, could we add another playwright check where we check that the same image used multiple times as an avatar for different `chat_message` doesn't cause any errors?

Reply:
oki 👍 I added one more e2e check that reuses the same local image. Does that work for checking this?

Reply:
yes, that covers exactly that case :)